src/simplewallet/simplewallet.cpp

```diff
@@ -6806,6 +6821,13 @@ int main(int argc, char* argv[])
   else
   {
     tools::signal_handler::install([&w](int type) {
+#ifdef HAVE_READLINE
+      if (simplewallet_suspend_readline::is_in_password())
```

Comment:
Hmm, good point about the race. However, it would need the user to input ^C really fast after sending a command, so it pretty unlikely.
This is kinda moot anyway, since jtgrassie mentioned fixing this in the read_from_tty code, which is better anyway and will moot my patch.

Reply:
I have been playing with the read_from_tty but am not yet able to properly fix. Using select with various timeout options and checking its errno for E_INTR on return -1, does work but it blocks the displaying of the password prompt. I think this is the place to fix so it's universal (i.e. not tied to readline) just I haven't found a solution yet! Sorry.

Reply:
Been messing around with this. Would love some direction / help.
```
diff --git a/src/common/password.cpp b/src/common/password.cpp
index ef026c97..182d9f0b 100644
--- a/src/common/password.cpp
+++ b/src/common/password.cpp
@@ -40,6 +40,7 @@
 #else
 #include <termios.h>
 #include <unistd.h>
+#include <sys/select.h>
 #endif
 
 #ifdef HAVE_READLINE
@@ -126,13 +127,29 @@ namespace
     return ch;
   }
 
+  static fd_set fds;
+
   bool read_from_tty(epee::wipeable_string& aPass)
   {
     static constexpr const char BACKSPACE = 127;
+    struct timeval t;
+    t.tv_sec = 1;
+    t.tv_usec = 0;
+    FD_ZERO(&fds);
+    FD_SET(STDIN_FILENO, &fds);
+
 
     aPass.reserve(tools::password_container::max_password_size);
     while (aPass.size() < tools::password_container::max_password_size)
     {
+      int sr = select(FD_SETSIZE, &fds, NULL, NULL, &t);
+
+      if(sr == -1)
+        return false;
+
+      if(sr == 0)
+        continue;
+
       int ch = getch();
       if (EOF == ch)
       {
```